src/rgw/rgw_op.cc

```diff
@@ -3257,6 +3257,34 @@ void RGWCreateBucket::execute()
         return;
       }
     }
+
+    auto old_acl_map = old_policy.get_acl().get_grant_map();
+    auto specified_acl_map = policy.get_acl().get_grant_map();
+
+    if (old_acl_map.size() != specified_acl_map.size()) {
+      op_ret = -EEXIST;
+      return;
+    }
+
+    std::map<string, ACLGrant>::iterator it = old_acl_map.begin();
+    while (it != old_acl_map.end()) {
+      std::map<string, ACLGrant>::iterator it_spec =
+          specified_acl_map.find(it->first);
+      if (it_spec != specified_acl_map.end()) {
+        bufferlist bl_old;
+        bufferlist bl_spec;
+        it_spec->second.encode(bl_spec);
+        it->second.encode(bl_old);
+        if (bl_spec.to_str() != bl_old.to_str()) {
+          op_ret = -EEXIST;
+          return;
+        }
+      } else {
+        op_ret = -EEXIST;
+        return;
+      }
+      it++;
+    }
```

Comment:
could we move this whole block of code into rgw_acl.cc and expose it as operator==?
```c++
bool operator==(const RGWAccessControlPolicy& lhs, const RGWAccessControlPolicy& rhs);
```